### PR TITLE
Swagger api

### PR DESCRIPTION
**Pull request by prioux**
This is the initial merging of the Swagger API information pages, with a very limited subset of APIs being described.

I want to merge this right away because of two significant features I had to implement as an aside to this project:

* a revamp of the ToolConfig information pages for users
* the addition of the `.for_api()` method for white listing attributes suitable for APIs

My plan is to keep updating the JSON Swagger API descriptor and go through all controllers and properly identify the attributes made available for `.for_api()`; I'd really like the help of the rest of the CBRAIN devs though!

### CODE REVIEW

File: BrainPortal/public/swagger/cbrain-4.5.1-swagger.yaml [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+        - name: login
+          in: query
+          description: The CBRAIN user's username
+          required: true
+          type: string
+        - name: password
+          in: query
+          description: The CBRAIN user's password
+          required: true
+          type: string
```
Comment (crocodoyle) [inline BrainPortal/public/swagger/cbrain-4.5.1-swagger.yaml]: In Swagger is there no password type? Could this have a security impact if users are trying out our API with their passwords?

File: BrainPortal/public/swagger/cbrain-4.5.1-swagger.yaml [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+        description: UNIX-style username.
+      full_name:
+        type: string
+        description: Full name of the user.
+      email:
+        type: string
+        description: email address of the user.
+      type:
+        type: string
+        description: class of the user, one of CorAdmin,AdminUser, SiteManager or NormalUser.
```
Comment (crocodoyle) [inline BrainPortal/public/swagger/cbrain-4.5.1-swagger.yaml]: typo

**Comment by crocodoyle** [comment]
ToolConfig page looks good to me

File: BrainPortal/app/controllers/portal_controller.rb [hunk 3]
```diff
@@ -361,6 +363,23 @@ def search
     @results = @search.present? ? ModelsReport.search_for_token(@search, current_user) : {}
   end
 
+  # A HTML GET request produces a SwaggerUI information page.
+  # A JSON GET request sends the swagger specification.
+  def swagger
+    # Find latest JSON swagger spec.
+    # FIXME sort() will break when comparing versions...
```
Comment (prioux) [inline BrainPortal/app/controllers/portal_controller.rb]: Nope; it is TO BE fixed.

Comment (prioux) [inline BrainPortal/app/controllers/portal_controller.rb]: Not very important right now, until we have many many API files to distribute... and right now we're still working on writing the first one!

File: BrainPortal/public/swagger/cbrain-4.5.0-swagger.json [hunk 4]
```diff
@@ -0,0 +1,204 @@
+{
```
Comment (prioux) [inline BrainPortal/public/swagger/cbrain-4.5.0-swagger.json]: Yes, that's the point. It's the description for THIS version of CBRAIN. We'll change the filename and keep them in sync as needed. But for the moment, anyway, we're still working on the first one.

File: BrainPortal/public/swagger/cbrain-4.5.0-swagger.json [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+                    },
+                    "401": {
+                        "description": "No session created yet."
+                    }
+                }
+            }
+        },
+        "/users/{id}": {
+            "get": {
+                "summary": "heelo",
```
Comment (prioux) [inline BrainPortal/public/swagger/cbrain-4.5.0-swagger.json]: Yes.

**Comment by natacha-beck** [comment]
This is a preliminary version, I just merge it.

**Pull request merged**